src/Artifact.php

```diff
@@ -303,26 +353,21 @@ protected function cleanup(): void {
    * @throws \Exception
    */
   protected function doPush(): void {
-    if (!$this->gitRemoteExists($this->src, $this->remoteName)) {
-      $this->gitAddRemote($this->src, $this->remoteName, $this->dst);
-    }
-
     try {
-      $this->gitPush(
-            $this->src,
-            $this->artifactBranch,
-            $this->remoteName,
-            $this->dstBranch,
-            $this->mode === self::modeForcePush()
-        );
-      $this->sayOkay(sprintf('Pushed branch "%s" with commit message "%s"', $this->dstBranch, $this->message));
+      $options = $this->mode === self::modeForcePush() ? ['--force'] : [];
```

Comment:
@tannguyen04 
`Artifact` class should not know anything about inner workings of git - this is why we provide our own proxy class to wrap around `git-php`: any specific git actions, like performing a force-push should be a matter of calling a method `forcePush()` , not passing a git-specific flag.

Please update the code of the Artifact class to be clear of git CLI flags.